### PR TITLE
FDIS-16: Logging improvement

### PR DESCRIPTION
**Pull request by shorasul96**
### Purpose
_Logging improvement._

### Approach
_All logs are written according the rule [Logging](https://wiki.folio.org/display/DD/Logging)_
_Configured pom.xml for enabling log4j2_
_Added logs into conditional cases_
_Added logs into exceptional cases_

### CODE REVIEW

File: src/main/java/org/folio/spring/SpringContextUtil.java [hunk 1]
```diff
@@ -40,22 +43,28 @@ public static void init(Vertx vertx, Context context) {
...
+      log.info("init:: configClassName != null");
       try {
         Class<?> springConfigClass = Class.forName(configClassName);
         AnnotationConfigApplicationContext springContext = createBaseContext(vertx, context);
         springContext.register(springConfigClass);
         springContext.refresh();
         context.put(SPRING_CONTEXT_KEY, springContext);
       } catch (ClassNotFoundException e) {
-        throw new IllegalStateException("Failed to load configuration class " + configClassName);
+        log.warn("Failed to load configuration class, msg: " + e.getMessage());
```
Comment (Shans-Kaluhin) [inline src/main/java/org/folio/spring/SpringContextUtil.java:57]:
```suggestion
        log.warn("Failed to load configuration class, msg: {}", e.getMessage());
```

File: src/main/java/org/folio/spring/SpringContextUtil.java [hunk 2]
```diff
@@ -9,20 +9,23 @@
...
+@Log4j2
 public class SpringContextUtil {
 
   private static final String SPRING_CONFIGURATION = "spring.configuration";
   private static final String SPRING_CONTEXT_KEY = "springContext";
   private static final String DEFAULT_CONFIGURATION_PACKAGE = "org.folio.spring.config";
 
-  private SpringContextUtil() {}
+  private SpringContextUtil() {
+  }
```
Comment (Shans-Kaluhin) [inline src/main/java/org/folio/spring/SpringContextUtil.java:28]: Leave it as it was
```suggestion
  private SpringContextUtil() {}
```

File: pom.xml [hunk 3]
```diff
@@ -21,6 +21,8 @@
   <properties>
     <vertx.version>4.3.4</vertx.version>
     <spring.version>5.3.23</spring.version>
+    <lombok.version>1.18.26</lombok.version>
+    <log4j.version>2.19.0</log4j.version>
```
Comment (Shans-Kaluhin) [inline pom.xml:25]:
```suggestion
    <log4j.version>2.20.0</log4j.version>
```

File: src/main/java/org/folio/spring/SpringContextUtil.java [hunk 4]
```diff
@@ -40,22 +42,28 @@ public static void init(Vertx vertx, Context context) {
...
+      log.info("init:: configClassName != null");
       try {
         Class<?> springConfigClass = Class.forName(configClassName);
         AnnotationConfigApplicationContext springContext = createBaseContext(vertx, context);
         springContext.register(springConfigClass);
         springContext.refresh();
         context.put(SPRING_CONTEXT_KEY, springContext);
       } catch (ClassNotFoundException e) {
-        throw new IllegalStateException("Failed to load configuration class " + configClassName);
+        log.warn("Failed to load configuration class, msg: {}" + e.getMessage());
```
Comment (Shans-Kaluhin) [inline src/main/java/org/folio/spring/SpringContextUtil.java:56]:
```suggestion
        log.warn("Failed to load configuration class, msg: {}",  e.getMessage());
```

File: src/main/java/org/folio/spring/SpringContextUtil.java [hunk 5]
```diff
@@ -40,22 +42,28 @@ public static void init(Vertx vertx, Context context) {
...
+        throw new IllegalStateException();
       }
     } else if (defaultConfiguration != null) {
+      log.info("init:: defaultConfiguration != null");
```
Comment (Shans-Kaluhin) [inline src/main/java/org/folio/spring/SpringContextUtil.java:60]: maybe something like this? 
```suggestion
      log.info("init:: Attempts to create spring context by defaultConfiguration");
```

File: src/main/java/org/folio/spring/SpringContextUtil.java [hunk 6]
```diff
@@ -40,22 +42,28 @@ public static void init(Vertx vertx, Context context) {
    */
   public static void init(Vertx vertx, Context context, Class<?> defaultConfiguration) {
     String configClassName = context.config().getString(SPRING_CONFIGURATION);
+    log.debug("init:: Attempts to create spring base context");
+
     if (configClassName != null) {
+      log.info("init:: configClassName != null");
```
Comment (Shans-Kaluhin) [inline src/main/java/org/folio/spring/SpringContextUtil.java:48]: maybe something like this? 
```suggestion
      log.info("init:: Attempts to create spring context by configClassName");
```

File: src/main/java/org/folio/spring/SpringContextUtil.java (hunk 5 above)
Comment (Shans-Kaluhin) [reply src/main/java/org/folio/spring/SpringContextUtil.java:60]: Ok, than line 66 should be like `init:: configClassName & defaultConfiguration == null`

**Review by Shans-Kaluhin — APPROVED** [review]

**Review by hasanepam — APPROVED** [review]

**Pull request merged**